modules/images-referencing-images-imagestreams.adoc

```diff
@@ -0,0 +1,175 @@
+// Module included in the following assemblies:
+// * openshift_images/tagging-images
+
+[id="images-referencing-images-imagestreams_{context}"]
+= Referencing images in imagestreams
+
+You can use tags to reference images in imagestreams using the following
+reference types.
+
+.Imagestream reference types
+[width="50%",options="header"]
+|===
+|Reference type |Description
+
+|`ImageStreamTag`
+|An `ImageStreamTag` is used to reference or retrieve an image for a given
+imagestream and tag.
+
+|`ImageStreamImage`
+|An `ImageStreamImage` is used to reference or retrieve an image for a given
+imagestream and image name.
+
+|`DockerImage`
+|A `DockerImage` is used to reference or retrieve an image for a given external
+registry. It uses standard Docker _pull specification_ for its name.
+|===
+
+When viewing example imagestream definitions you may notice they contain
+definitions of `ImageStreamTag` and references to `DockerImage`, but nothing
+related to `ImageStreamImage`.
+
+This is because the `ImageStreamImage` objects are automatically created in
+{product-title} when you import or tag an image into the imagestream. You
+should never have to explicitly define an `ImageStreamImage` object in any
+imagestream definition that you use to create imagestreams.
+
+.Procedure
+
+* To reference an image for a given imagestream and tag, use `ImageStreamTag`:
++
+----
+<image_stream_name>:<tag>
+----
+
+* To reference an image for a given imagestream and image name, use
+`ImageStreamImage`:
++
+----
+<image_stream_name>@<id>
+----
++
+The `<id>` is an immutable identifier for a specific image, also called a
+digest.
+
+* To reference or retrieve an image for a given external
+registry, use `DockerImage`:
++
+----
+openshift/ruby-20-centos7:2.0
+----
++
+[NOTE]
+====
+When no tag is specified, it is assumed the *latest* tag is used.
+====
++
+You can also reference a third-party registry:
++
+----
+registry.redhat.io/rhel7:latest
+----
++
+Or an image with a digest:
++
+----
+centos/ruby-22-centos7@sha256:3a335d7d8a452970c5b4054ad7118ff134b3a6b50a2bb6d0c07c746e8986b28e
+----
+
+* You can view an image's object definition by retrieving an `ImageStreamImage`
+definition using the imagestream name and ID:
```

Comment:
the `get imagestreamimage` can't be used in 4.x see https://bugzilla.redhat.com/show_bug.cgi?id=1690236